hackathon_site/dashboard/frontend/src/pages/IncidentForm/IncidentForm.tsx

```diff
@@ -0,0 +1,463 @@
+import React, { useEffect, useRef } from "react";
+import styles from "./IncidentForm.module.scss";
+import {
+    Typography,
+    Card,
+    CardContent,
+    RadioGroup,
+    FormControlLabel,
+    Radio,
+    Select,
+    MenuItem,
+    Button,
+    TextField,
+    FormControl,
+    InputLabel,
+    FormHelperText,
+    makeStyles,
+} from "@material-ui/core";
+import { useHistory, useLocation } from "react-router-dom";
+import { Formik, Field, FormikValues } from "formik";
+import * as Yup from "yup";
+import Header from "components/general/Header/Header";
+import ArrowLeft from "../../assets/images/icons/arrow-left-solid.svg";
+import { displaySnackbar } from "slices/ui/uiSlice";
+import { useDispatch } from "react-redux";
+
+export const INCIDENT_ERROR_MSG = {
+    state: "Please select an option",
+    qtyEmpty: "Please indicate the quantity",
+    qtyZero: "Cannot submit incident form for 0 items",
+    what: "Please indicate what happened to the hardware",
+    when: "Please indicate when this occurred",
+    where: "Please indicate where this occurred",
+};
+
+const useStyles = makeStyles({
+    typographyLabel: {
+        paddingBottom: "10px", // Adjust the value as per your requirement
+    },
+});
+
+const capitalizeFirstLetter = (str: string) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+const validationSchema = Yup.object({
+    state: Yup.string().required(INCIDENT_ERROR_MSG.state),
+    qty: Yup.string()
+        .test("not-zero", INCIDENT_ERROR_MSG.qtyZero, (value) => value !== "0")
+        .required(INCIDENT_ERROR_MSG.qtyEmpty),
+    what: Yup.string().required(INCIDENT_ERROR_MSG.what),
+    when: Yup.string().required(INCIDENT_ERROR_MSG.when),
+    where: Yup.string().required(INCIDENT_ERROR_MSG.where),
+});
+
+const initialValues = {
+    state: "",
+    qty: "",
+    what: "",
+    when: "",
+    where: "",
+};
+
+const createQuantityList = (number: number) => {
+    let entry = [];
+    if (number === 0) {
+        entry.push(
```

Comment:
This can be refactored since the 0 case is unnecessary now

Reply:
thanks @KarandeepLubana !